chrome-extension/sidebar/sidebar.js

```diff
@@ -48,6 +48,307 @@ const htmlToElement = (html) => {
 };
 
+// Begin code for sorting course list
+document.addEventListener('change', function(event) {
+  if (event.target.matches('.size-dropdown')) {
+    handleDropdownChange(event);
+  }
+});
+
+// padding: 6px 10px;
+// padding: 4px 10px;
+const dropdownTemplate = `
+<div style="display: inline-block; margin-left: -20px; margin-right: 5px;">
+  <select class="size-dropdown" style="height: 34px; transform: translateY(3%); font-size: 14px; padding: 6px 12px;">
+    <option value="A">Sort By Course # (Default)</option>
+    <option value="B">Sort By Seats Available</option>
+    <option value="C">Sort By Most-Common Grade</option>
+    <option value="D">Sort By Popularity</option>
+    <option value="E">Sort By Units</option>
+  </select>
+</div>
+`;
+
+const handleDropdownChange = (event) => {
+  if (event.target.value === "A") {
+    location.reload();
+  }
+  if (event.target.value === "B") {
+    const buttons = document.querySelectorAll('.action-sections.btn.btn-default');    
+    buttons.forEach(button => button.click());
+    setTimeout(sortBySeatsAvailable, 1000);
+  }
+  if (event.target.value === "C") {
+    sortByMostCommonGrade();
+  }
+  if (event.target.value === "D") {
+    sortByPopularity();
+  }
+  if (event.target.value === "E") {
+    sortByUnits();
+  }
+};
+
+const sortBySeatsAvailable = () => {
+  const courseSizeDict = {};
+
+  // Get the main container with class 'course-list-results'
+  let courseListResults = document.querySelector('.course-list-results');
+  let courseDivs = Array.from(courseListResults.firstElementChild.children);
+
+  for (let courseDiv of courseDivs) {
+    let courseName = courseDiv.querySelector('a').getAttribute('name');
+    
+    // Navigate to the correct div
+    let targetDiv = courseDiv.querySelector('div:nth-child(2) > div:nth-child(3)');
+    let trElements = targetDiv.querySelectorAll('tr');
+    
+    let sum = 0;
+    trElements.forEach(tr => {
+        let lastChildText = tr.lastElementChild.innerText;
+        sum += parseFloat(extractDifference(lastChildText)) || 0; // Ensure the text is converted to a number
+    });
+    
+    courseSizeDict[courseName] = sum;
+  }
+
+  courseDivs.sort((a, b) => {
+    let nameA = a.querySelector('a').getAttribute('name');
+    let nameB = b.querySelector('a').getAttribute('name');
+    return courseSizeDict[nameB] - courseSizeDict[nameA];
+  });
+
+  console.log(courseSizeDict); // For debugging purposes, display the updated dictionary
+  console.log(courseDivs); // For debugging purposes, display the updated dictionary
+
+
+  courseListResults.firstElementChild.innerHTML = '';
+  courseDivs.forEach(courseDiv => {
+      courseListResults.firstElementChild.appendChild(courseDiv);
+  });
+
+  const buttons2 = document.querySelectorAll('.action-sections.btn.btn-default.pull-right');
+  buttons2.forEach(button => button.click());
+
+  document.querySelector('.size-dropdown').value = 'B';
+};
+
+function extractDifference(text) {
```

Comment:
you have inconsistent spacing, set your IDE to use 2spaces per indent instead of the 4 that this function is using